plugins/slicer/MONAILabel/MONAILabel.py

```diff
@@ -1280,26 +1340,30 @@ def onSaveLabel(self):
         try:
             qt.QApplication.setOverrideCursor(qt.Qt.WaitCursor)
             segmentationNode = self._segmentNode
-            labelmapVolumeNode = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLLabelMapVolumeNode")
-            slicer.modules.segmentations.logic().ExportVisibleSegmentsToLabelmapNode(
-                segmentationNode, labelmapVolumeNode, self._volumeNode
-            )
-
             segmentation = segmentationNode.GetSegmentation()
             totalSegments = segmentation.GetNumberOfSegments()
             segmentIds = [segmentation.GetNthSegmentID(i) for i in range(totalSegments)]
 
+            # remove background and scribbles labels
             label_info = []
+            save_segment_ids = vtk.vtkStringArray()
             for idx, segmentId in enumerate(segmentIds):
                 segment = segmentation.GetSegment(segmentId)
-                if segment.GetName() in ["foreground_scribbles", "background_scribbles"]:
+                if segment.GetName() in ["background", "foreground_scribbles", "background_scribbles"]:
                     logging.info(f"Removing segment {segmentId}: {segment.GetName()}")
                     segmentationNode.RemoveSegment(segmentId)
                     continue
 
+                save_segment_ids.InsertNextValue(segmentId)
```

Comment:
Potentially, segment removal can be dropped now that `save_segment_ids` is defined? I'm not sure that the segment removal is necessary for any further downstream tasks.

```suggestion
                if segment.GetName() not in ["background", "foreground_scribbles", "background_scribbles"]:
                    save_segment_ids.InsertNextValue(segmentId)
```

Reply:
Agreed! I am removing `RemoeSegment` in latest commit.